File: semestr-8/Jezyk C#/alexander/PCS/2023s/ID06TC1/ManualDrawing/ManualDrawing/Shape.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Drawing;

namespace ManualDrawing
{
    abstract class Shape
    {
        public delegate void SI(int v);
        public void ParseParameters(string parameters, params SI[] si)
        {
            string[] split = parameters.Split(" \t;".ToCharArray(), StringSplitOptions.RemoveEmptyEntries);
            int size = si.Length;
            if (size != split.Length) throw new ArgumentException("Invalid array size");
            for (int i = 0; i < size; ++i) si[i](int.Parse(split[i]));
            //foreach (SI f in si) f(int.Parse(split[i]));
        }
        /*
        public int[] ParseParameters(string parameters,int expectedSize)
        {
            string[] split = parameters.Split(" \t;".ToCharArray(), StringSplitOptions.RemoveEmptyEntries);
            int size = split.Length;
            if (size != expectedSize) throw new ArgumentException("Invalid array size");
            //if (size != expectedSize) return null;
            int[] param = new int[size];
            for (int i = 0; i < size; ++i) param[i] = int.Parse(split[i]);
            return param;
        }
        */
        public abstract void Draw(Graphics g);
    }
}

File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/Archiwum/2022s/IZ08IO2/BridgeTestApp/src/Shape.java

import java.awt.Graphics;

public abstract class Shape
{
   protected int[] tb;
   public Shape(String parameters,int count)
   {
      String[] vals=parameters.split("[\\s\t;]+");
      if(vals.length!=count) throw new IllegalArgumentException("Wrong parameters count");
      tb=new int[vals.length];
      for(int i=0;i<vals.length;++i) tb[i]=Integer.parseInt(vals[i]);
   }
   public abstract void Draw(Graphics g);
   public abstract String toXml();
}
